packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,217 @@
+| defaultValue  | `string[]`                  | No       | `[]`        | The initial value of the checkbox group                                                                                            |
+| value         | `string[]`                  | No       | `[]`        | The value of the checkbox, if present will act as controlled component                                                             |
+| onChange      | `(value: string[]) => void` | No       | `undefined` | The function to be called when any checkbox's state changes                                                                        |
+| name          | `string`                    | No       | `undefined` | The name of the checkbox group, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name) |
+| isDisabled    | `boolean`                   | No       | `false`     | Control whether the checkbox group is disabled or not.                                                                             |
+| hasError     | `boolean`                   | No       | `false`     | Control whether the checkbox group is invalid or not.                                                                              |
+| showOptionalLabel    | `boolean`                   | No       | `false`     | Shows optional label in checkbox group header |
+
+### Examples:
+
+#### Basic
+
+```tsx
+<CheckboxGroup
+  label="Developers"
+  helpText="Pick blade developers"
+  defaultValue={['anurag', 'kamlesh']}
+>
+  <Checkbox value="anurag">Anurag</Checkbox>
+  <Checkbox value="kamlesh">Kamlesh</Checkbox>
+  <Checkbox value="chaitanya">Chaitanya</Checkbox>
+</CheckboxGroup>
+```
+
+#### Controlled, Uncontrolled
+
+```tsx
+const Controlled = () => {
+  const [selected, setSelected] = React.useState(['anurag', 'kamlesh']);
+
+  return (
+    <CheckboxGroup 
+      label="Developers (controlled)" 
+      value={selected} 
+      onChange={setSelected}
+    >
+      <Checkbox value="anurag">Anurag</Checkbox>
+      <Checkbox value="kamlesh">Kamlesh</Checkbox>
+      <Checkbox value="chaitanya">Chaitanya</Checkbox>
+    </CheckboxGroup>
+  );
+};
+
+const Uncontrolled = () => {
+  return (
+    <CheckboxGroup
+      label="Developers (uncontrolled)"
+      defaultValue={['anurag', 'chaitanya']}
+    >
+      <Checkbox value="anurag">Anurag</Checkbox>
+      <Checkbox value="kamlesh">Kamlesh</Checkbox>
+      <Checkbox value="chaitanya">Chaitanya</Checkbox>
+    </CheckboxGroup>
+  );
+};
+```
+
+### Note on CheckboxGroup
+
+For properties `isDisabled`, `hasError` we want the CheckboxGroup to pass down these props to all the Checkboxes inside of it.
+
+For showing optional label in `CheckboxGroup` we will introduce a prop `showOptionalLabel` which will only render a `(optional)` tag in the CheckboxGroupLabel,
```

Comment:
I think we can add a screenshot here from design of how the `optional` label will look so that it's clearer.

Reply:
Added